### PR TITLE
Plfm 5688

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/table/TableUpdateTransactionRequest.json [hunk 1]
```diff
@@ -16,9 +16,9 @@
 				"$ref": "org.sagebionetworks.repo.model.table.TableUpdateRequest"
 			}
 		},
-		"versionRequest": {
-			"description": "An optional parameter used to create a new version of table that includes all of the changes from this request.",
-			"$ref": "org.sagebionetworks.repo.model.table.VersionRequest"
+		"snapshotRequest": {
```
Comment (john-hill) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/table/TableUpdateTransactionRequest.json]: boolean createSnapshot
SnapshotRequest snapshotOptions

File: services/repository/src/main/java/org/sagebionetworks/repo/web/service/metadata/TableEntityMetadataProvider.java [hunk 2]
```diff
@@ -30,10 +30,10 @@ public void entityDeleted(String deletedId) {
...
 	public void entityUpdated(UserInfo userInfo, TableEntity entity, boolean wasNewVersionCreated) {
-		tableEntityManager.setTableSchema(userInfo, entity.getColumnIds(), entity.getId());
 		if(wasNewVersionCreated) {
-			tableEntityManager.bindCurrentEntityVersionToLatestTransaction(entity.getId());
+			throw new IllegalArgumentException("A table version can only be created by creating a table snapshot.");
 		}
+		tableEntityManager.setTableSchema(userInfo, entity.getColumnIds(), entity.getId());
 	}
 
 	@Override
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/service/metadata/TableEntityMetadataProvider.java:41]: newly created table should be "in progress"

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/NodeManagerImpl.java [hunk 3]
```diff
@@ -62,6 +63,8 @@
  */
 public class NodeManagerImpl implements NodeManager {
 
+	public static final String IN_PROGRESS = "in-progress";
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/NodeManagerImpl.java]: "in progress"

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableEntityManager.java [hunk 4]
```diff
@@ -290,7 +284,7 @@ public TableUpdateResponse updateTable(ProgressCallback callback, UserInfo userI
 	 * @param transactionId
 	 * @return The version number of the newly created version.
 	 */
-	public long createNewVersionAndBindToTransaction(UserInfo userInfo, String tableId, VersionRequest versionRequest,
+	public long createSnapshotBindToTransaction(UserInfo userInfo, String tableId, SnapshotRequest snapshotRequest,
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableEntityManager.java]: andBind

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableEntityManagerImpl.java [hunk 5]
```diff
@@ -956,4 +944,26 @@ void linkVersionToTransaction(String tableIdString, long version, long transacti
...
+		ValidateArgument.required(tableIdString, "TableId");
+		ValidateArgument.required(request, "request");
+		Long tableId = KeyFactory.stringToKey(tableIdString);
+		IdAndVersion idAndVersion = IdAndVersion.newBuilder().setId(tableId).build();
+		// Validate the user has permission to edit the table
+		tableManagerSupport.validateTableWriteAccess(userInfo, idAndVersion);
+		
+		Optional<Long> lastTransactionNumber = tableRowTruthDao.getLastTransactionId(tableIdString);
+		if(!lastTransactionNumber.isPresent()) {
+			throw new IllegalArgumentException("No transactions found for table: "+tableId);
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableEntityManagerImpl.java]: This table does not have a schema so a snapshot cannot be created. //comment about transactions

**Review by zimingd — APPROVED** [review]

**Pull request merged**